api/envoy/config/route/v3/route_components.proto

```diff
@@ -875,6 +875,26 @@ message RouteAction {
     google.protobuf.Duration grpc_timeout_header_offset = 3;
   }
 
+  message EarlyDataOptions {
+    // The request methods allowed to be sent over early data.
+    // If empty, no request is allowed to use early data.
+    repeated string allowed_methods = 1 [(validate.rules).repeated = {
+      items {
+        string {
+          in: "GET"
```

Comment:
In the E-M weekly meeting we discussed about not relying on :method types, instead, an Envoy header: x-envoy-safe-for-early-data.

How about replacing early_data_options here with a bool which enables checking the presence of `x-envoy-safe-for-early-data` header? Either this bool or `early_data_allows_safe_requests` should take effect.

@goaway

Reply:
Since the use case for customized early data option is unclear, I will make the whole thing an extension with bool early_data_allows_safe_requests config knob in the default config.

Reply:
For EM's purposes, being able to specify this with a heady would be nice, since we can then expose an API that allows callers to declare whether requests are idempotent (i.e. safe for early data) or not.

Reply:
SGTM

Reply:
extension added. @markdroth PTAL.